src/compiler/parser.ts

```diff
@@ -3183,6 +3183,7 @@ namespace ts {
                     if (sourceFile.languageVariant !== LanguageVariant.JSX) {
                         return parseTypeAssertion();
                     }
+                    debugger;

```

Comment:
We could add a lint for this. No lies. Alongside 'no-null' we can have 'no-debugger'.